pkg/webhook/stormservice_webhook.go

```diff
@@ -0,0 +1,244 @@
+					MountPath: SidecarModelMountPath,
+				},
+			},
+			LivenessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Path: SidecarHealthPath,
+						Port: intstr.FromInt(SidecarPort),
+					},
+				},
+				InitialDelaySeconds: 3,
+				PeriodSeconds:       2,
+			},
+			ReadinessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Path: SidecarReadyPath,
+						Port: intstr.FromInt(SidecarPort),
+					},
+				},
+				InitialDelaySeconds: 5,
+				PeriodSeconds:       10,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("100m"),
+					corev1.ResourceMemory: resource.MustParse("256Mi"),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("500m"),
+					corev1.ResourceMemory: resource.MustParse("512Mi"),
+				},
+			},
+		}
+
+		// Inject sidecar at the beginning
+		role.Template.Spec.Containers = append(
+			[]corev1.Container{runtimeContainer},
+			role.Template.Spec.Containers...,
+		)
+
+		// Ensure volume exists
+		if !hasVolume(role.Template.Spec.Volumes, SidecarVolumeName) {
+			role.Template.Spec.Volumes = append(role.Template.Spec.Volumes, corev1.Volume{
+				Name: SidecarVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			})
+		}
+	}
+}
+
+// inferEngineType infers the inference engine based on container image names
+func (r *StormServiceCustomDefaulter) inferEngineType(containers []corev1.Container) string {
+	for _, c := range containers {
+		img := strings.ToLower(c.Image)
+		if strings.Contains(img, "vllm") {
+			return "vllm"
+		}
+		if strings.Contains(img, "text-generation-inference") || strings.Contains(img, "tgi") {
+			return "tgi"
+		}
+		if strings.Contains(img, "triton") {
+			return "triton"
+		}
+		if strings.Contains(img, "llama") && strings.Contains(img, "cpp") {
+			return "llamacpp"
+		}
+	}
+	return "unknown"
+}
+
+// containsContainer checks if a container with the given name exists
+func containsContainer(containers []corev1.Container, name string) bool {
+	for _, c := range containers {
+		if c.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+// hasVolume checks if a volume with the given name exists
+func hasVolume(volumes []corev1.Volume, name string) bool {
+	for _, v := range volumes {
+		if v.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+// Sidecar injection constants
+const (
+	SidecarInjectionAnnotation = "stormservice.orchestration.aibrix.ai/sidecar-injection"
+	SidecarName                = "aibrix-runtime"
+	SidecarImage               = "aibrix-container-registry-cn-beijing.cr.volces.com/aibrix/runtime:v0.3.0"
+	SidecarCommand             = "aibrix_runtime"
+	SidecarPort                = 8080
+	SidecarModelMountPath      = "/models"
+	SidecarVolumeName          = "model-hostpath"
```

Comment:
We have not implement the features to resue NFS between nodes, or share single host volumes (security concern) can we make it configurable later?